### PR TITLE
Update PULL_REQUEST_TEMPLATE to include an API spec change in the checklist.

### PR DESCRIPTION
**Pull request by dblock**
Coming from https://github.com/opensearch-project/opensearch-api-specification/issues/387, update PR template to include a checkbox that requests that all API changes be documented in the OpenAPI spec used to generate clients.

### CODE REVIEW

**Review by reta — APPROVED** [review]

**Comment by reta** [comment]
Build should be fixed by https://github.com/opensearch-project/opensearch-oci-object-storage/pull/62

**Comment by reta** [comment]
@dblock could you rebase please? thank you

**Review by reta — APPROVED** [review]

**Comment by andrross** [comment]
@dblock Eh, I'm not sure about this PR. I dislike unnecessary boilerplate, and it strikes me as unlikely that a repository plugin would ever need to implement a user-facing API that would require a change to the specification.

**Comment by dblock** [comment]
@andrross If you don't think this plugin will ever implement a REST API then we can just close it. Your call!

File: .github/PULL_REQUEST_TEMPLATE.md
Comment (andrross) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Can we use the leaner version of this checklist that is here?

https://github.com/opensearch-project/OpenSearch/blob/ac78f0a4dc39f5b4be32adc8bf54999b2b44cd14/.github/pull_request_template.md?plain=1#L16-L18

Comment (dblock) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: This is a standard template though, do you really want me to go and change it everywhere?

**Comment by dblock** [comment]
Can any current maintainers please merge this?

@samuel-oci @anasalkouz @VachaShah?

**Review by andrross — APPROVED** [review]

**Pull request merged**
